### PR TITLE
Add Preferred languages to Application Registration Configuration

### PR DESCRIPTION
**Pull request by spwitt**
Add `preferredLanguages` to `ApplicationRegistrationConfiguration` APIs. Update screenshot.

**Site Issue**:
- https://github.com/FusionAuth/fusionauth-site/issues/1943

**Related Issue**:
- https://github.com/FusionAuth/fusionauth-issues/issues/1738

### CODE REVIEW

**Review by mooreds — CHANGES_REQUESTED** [review]
Can you please update this tutorial with a new screenshot? https://fusionauth.io/docs/v1/tech/guides/basic-registration-forms

Not sure if this should be documented here, but what does the preferred languages field look like? is it the same as the locale selector? How do I control what shows up there?

**Comment by spwitt** [comment]
@mooreds updated the guide screenshot.

The selection uses https://fusionauth.io/docs/v1/tech/reference/data-types#locales

There is no way to configure which locales appear in the list similar to the existing preferred languages dropdown.

File: site/docs/v1/tech/apis/_application-request-body.adoc [hunk 1]
```diff
@@ -310,6 +310,12 @@ Determines if the `mobilePhone` field will be included on the registration form.
 [field]#application.registrationConfiguration.mobilePhone.required# [type]#[Boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.4.0#::
 Determines if the `mobilePhone` field is required when displayed on the registration form.
 
+[field]#application.registrationConfiguration.preferredLanguages.enabled# [type]#[Boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.44.0#::
+Determines if the `preferredLanguages` field will be included on the registration form.
```
Comment (mooreds) [inline site/docs/v1/tech/apis/_application-request-body.adoc:314]:
```suggestion
Determines if the `preferredLanguages` field will be included on the registration form. The field will display all available link:/docs/v1/tech/reference/data-types#locales[locales] and is not configurable.
```

**Review by mooreds — APPROVED** [review]
One small change requested.

**Comment by mooreds** [comment]
Or is this waiting for 1.45? Oops, looks like it is.

**Comment by mooreds** [comment]
Would be awesome if this made it into 1.47.

File: site/docs/v1/tech/apis/_application-request-body.adoc [hunk 2]
```diff
@@ -361,6 +361,12 @@ Determines if the `mobilePhone` field will be included on the registration form.
 [field]#application.registrationConfiguration.mobilePhone.required# [type]#[Boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.4.0#::
 Determines if the `mobilePhone` field is required when displayed on the registration form.
 
+[field]#application.registrationConfiguration.preferredLanguages.enabled# [type]#[Boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.44.0#::
```
Comment (robotdan) [inline site/docs/v1/tech/apis/_application-request-body.adoc:364]: 1.47.0, same for other places as well.

File: site/docs/v1/tech/apis/_application-request-body.adoc [hunk 3]
```diff
@@ -361,6 +361,12 @@ Determines if the `mobilePhone` field will be included on the registration form.
...
+Determines if the `preferredLanguages` field will be included on the registration form. The field will display all available link:/docs/v1/tech/reference/data-types#locales[locales] and is not configurable.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/_application-request-body.adoc:365]: None of the basic form values are really configurable? 

> and is not configurable.

Does this just mean you don't get to choose the values displayed? If so, this is not really accurate. You could build your own locale selector. Our default control uses `fusionAuth.locales()` in a `[#list` to iterate through all of the values, but you don't have to use this.

**Review by robotdan — APPROVED** [review]

**Comment by mooreds** [comment]
@spwitt super exciting! This is great for folks who are on the community version and want to make sure their initial emails are sent out in the correct language.

**Pull request merged**